internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -732,6 +769,11 @@ func resourceOrchestratedVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData,
 			}
 		}
 
+		if d.HasChange("sku_profile") {
```

Comment:
This block is `ForceNew` in the schema, why are we checking for updates?

Reply:
answered in the [above ](https://github.com/hashicorp/terraform-provider-azurerm/pull/27599#discussion_r1818675141)comment. Though it's defined `ForceNew`, its sub-properties can be updated changed through `Update` as long as `sku_profile` is not removed.